### DISCUSSION TITLE
Global command are not permitted to execute

### DISCUSSION THREAD

**Discussion by alirezaopmc** [johnlindquist/kit#1127] [Errors]
I have my own CLI written in js and installed using the command `npm install -g .` from inside of the package. My CLI command works fine in my terminal.

My CLI:
```bash
xcli <cmd> [args]
```

I only use one of the command like this:
```bash
xcli fix something
```

But xcli command cannot be found by kit. I provide the full path and it says not permitted.
How can I solve this issue?

This is the content of my script:

```ts
// Name: Fix VMESS
// Shortcut: cmd j

import "@johnlindquist/kit";


let vmess = await arg("Enter the vmess link: ");
let xcli = `/Users/alirezaopmc/.nvm/versions/node/v18.14.0/bin/xcli`;

await $`chmod +x ${xcli}`
await $`${xcli} fix ${vmess}`
// copy(stdout);
```

**Comment by johnlindquist** [comment]
@alirezaopmc My first suggestion is to run this:

![CleanShot 2023-02-16 at 04 27 52](https://user-images.githubusercontent.com/36073/219352922-1f44ec67-2076-4c4f-85fd-0bcec5a7b510.png)

Then you should be able to run it without using the full path.

## "Not Permitted"

Can you please copy/paste/screenshot the full error? I'd like to see it just to make sure it's not something else

## chmod

You shouldn't need the `chmod` line since your CLI is already executable

## quotes

Make sure to read this about using `$` (AKA "zx") with quotes in case that's part of the issue:

https://github.com/google/zx/blob/main/docs/quotes.md

**Comment by alirezaopmc** [comment]
Thank you for your detailed reply. This is the error log: https://termbin.com/7e96n

**Reply by johnlindquist** [reply]
@alirezaopmc Based on the error log here, it looks like your tool lives in a folder on your Desktop:

```
Error: EPERM: operation not permitted, open '/Users/alirezaopmc/Desktop/opmc/Dev/Xray/App/bin/index.js'
```

I think you'll need to grant Kit.app permission to access files on your Desktop:

![CleanShot 2023-02-16 at 12 43 25](https://user-images.githubusercontent.com/36073/219470457-c21b6276-f13d-4fbc-9275-7d4ef325454c.png)

> Node: You may need to quit Kit.app and re-open it for the security changes to take effect
